local-proxy/src/server.ts

```diff
@@ -66,10 +66,20 @@ app.post('/invoke', json(), async (req, res) => {
   const { path, handler, args } = req.body;
   registry.register(requestId);
   try {
-    await transpilePromise;
-    // Make sure we use tmpDir as absolute path
-    const mod = require(pathJoin(tmpDir, path));
-    const fn = mod[handler];
+    let fn: (...args: any[]) => any;
+    if (path === '@binaris/shift-db' && (handler === 'getVersioned' || handler === 'poll')) {
+      fn = require(path)[handler];
+    } else {
+      await transpilePromise;
+      const joinedDir = pathResolve(tmpDir, path);
+      if (!joinedDir.startsWith(tmpDir)) {
```

Comment:
Supposedly this check should be more robust for win32: https://github.com/binaris/shiftjs/blob/b1960e15c96075e0dcdce0b8c4797a6d9920673c/babel-macro/src/macro.ts#L105.

Reply:
That doesn't seem right..
What about `a/../../b`?

Reply:
relative is the result of `path.relative` so it shouldn't contain `..`. This is taken from [SO](https://stackoverflow.com/questions/37521893/determine-if-a-path-is-subdirectory-of-another-in-node-js), maybe node `path-is-inside` is better, although it has quite a different implementation.